schemas/v1.0/schema.yaml

```diff
@@ -0,0 +1,525 @@
+  The description of OpenAPI Initiative Arazzo v1.0.0 documents
+  without schema validation, as defined by https://spec.openapis.org/arazzo/v1.0.0
+type: object
+properties:
+  arazzo:
+    description: The version number of the Arazzo Specification
+    type: string
+    pattern: '^1\.\d+\.\d+(-.+)?$'
+  info:
+    $ref: '#/$defs/info'
+  sourceDescriptions:
+    description: A list of source descriptions such as Arazzo or OpenAPI
+    type: array
+    uniqueItems: true
+    minItems: 1
+    items:
+      $ref: '#/$defs/source-description-object'
+  workflows:
+    description: A list of workflows
+    type: array
+    uniqueItems: true
+    minItems: 1
+    items:
+      $ref: '#/$defs/workflow-object'
+  components:
+    $ref: '#/$defs/components-object'
+required:
+  - arazzo
+  - info
+  - sourceDescriptions
+  - workflows
+unevaluatedProperties: false
+$defs:
+  info:
+    $comment: https://spec.openapis.org/arazzo/v1.0.0#info-object
+    description: Provides metadata about the Arazzo description
+    type: object
+    properties:
+      title:
+        description: A human readable title of the Arazzo Description
+        type: string
+      summary:
+        description: A short summary of the Arazzo Description
+        type: string
+      description:
+        description: A description of the purpose of the workflows defined. CommonMark syntax MAY be used for rich text representation
+        type: string
+      version:
+        description: The version identifier of the Arazzo document (which is distinct from the Arazzo Specification version)
+        type: string
+    required:
+      - title
+      - version
+    $ref: '#/$defs/specification-extensions'
+    unevaluatedProperties: false
+  source-description-object:
+    $comment: https://spec.openapis.org/arazzo/v1.0.0#source-description-object
+    description: |-
+      Describes a source description (such as an OpenAPI description)
+      that will be referenced by one or more workflows described within
+      an Arazzo description
+    type: object
+    properties:
+      name:
+        description: A unique name for the source description
+        type: string
+        pattern: '^[A-Za-z0-9_\-]+$'
+      url:
+        description: A URL to a source description to be used by a workflow
+        type: string
+        format: uri-reference
+      type:
+        description: The type of source description
+        enum:
+          - arazzo
+          - openapi
+    required:
+      - name
+      - type
+    $ref: '#/$defs/specification-extensions'
+    unevaluatedProperties: false
+  workflow-object:
+    $comment: https://spec.openapis.org/arazzo/v1.0.0#workflow-object
+    description: Describes the steps to be taken across one or more APIs to achieve an objective
+    type: object
+    properties:
+      workflowId:
+        description: Unique string to represent the workflow
+        $dynamicAnchor: workflowId
+        type: string
+        pattern: '^[A-Za-z0-9_\-]+$'
```

Comment:
given the statement 
" it is RECOMMENDED to follow common programming naming conventions. SHOULD conform to the regular expression [A-Za-z0-9_\-]+." 
i am not sure if  the json schema should enforce good practices or not 

the issue over there is that spec that does not follow this , are valid as it is just a SHOULD , whereas leveraging this schema with strict validation it might be rejected ... not sure as pattern is not a mandatory to follow as well

Reply:
this is consistent with the OpenAPI Specification.  They enforce this pattern on component and schemas names and use the same language

Reply:
I don't see anything in the OpenAPI schema/spec that is like this.
The closest I see, `operationId`, does not specify a pattern.
Other patterns correspond to MUSTs, e.g. "All the fixed fields declared above are objects that MUST use keys that match the regular expression: `^[a-zA-Z0-9\.\-_]+$`." I may be overlooking something but I don't see any that don't.
I think that the schema should not reject values that are not forbidden, even if recommended against; they are still valid.

Reply:
I didn't exactly follow their pattern, which now that I look at it closer, I probably could update it to match 100%.

but they enumerate all the `components` names and then use `propertyNames` to enforce it. 
https://github.com/OAI/OpenAPI-Specification/blob/ba055ca00cbd735dceef4ed7c9db024cd9bfcd1a/schemas/v3.1/schema.yaml#L199

Reply:
EDIT: 
opps. completely misread the original comment from Francois and then interpreted your comment in the same context as my original interpretation. I see what you're referring to now

i suppose `stepId` and `workflowId` can be relaxed to `type: string` since a `MUST` is not defined

i will defer to the spec authors to comment here. @frankkilcommins